### PR TITLE
gui-wm/wayfire: bump to 0.7.3

### PR DESCRIPTION
**Pull request by m1027**
Closes:  https://bugs.gentoo.org/840978

### CODE REVIEW

File: gui-wm/wayfire/wayfire-0.7.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+		x11-libs/libxcb
+	)
+	system-wfconfig? (
+		>=gui-libs/wf-config-0.7.1
+		<gui-libs/wf-config-0.8.0
+	)
+	!system-wfconfig? ( !gui-libs/wf-config )
+	system-wlroots? (
+		>=gui-libs/wlroots-0.15.0:=[X?]
+		<gui-libs/wlroots-0.16.0:=
```
Comment (juippis) [inline gui-wm/wayfire/wayfire-0.7.3.ebuild:50]: Maybe this'd be simpler if you just dependend on the subslot?
```suggestion
		gui-libs/wlroots:0/15
```

File: gui-wm/wayfire/wayfire-0.7.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	media-libs/freetype:=[X?]
+	x11-libs/libdrm
+	x11-libs/gtk+:3=[wayland,X?]
+	x11-libs/cairo:=[X?,svg]
+	x11-libs/libxkbcommon:=[X?]
+	x11-libs/pixman
+	X? (
+		x11-base/xwayland
+		x11-libs/libxcb
+	)
```
Comment (juippis) [inline gui-wm/wayfire/wayfire-0.7.3.ebuild:42]:
```
meson.build:141:2: ERROR: Problem encountered: Cannot enable Xwayland in wayfire: wlroots has been built without Xwayland support
```
```
[ERROR] >>> Not all runs were successful.
atom: =gui-wm/wayfire-0.7.3, USE flags: 'X -gles -system-wfconfig system-wlroots'
atom: =gui-wm/wayfire-0.7.3, USE flags: 'X -gles system-wfconfig -system-wlroots'
atom: =gui-wm/wayfire-0.7.3, USE flags: 'X -gles system-wfconfig system-wlroots'
```
So `+X` seems to fail, apparently to a missing dependency.

File: gui-wm/wayfire/wayfire-0.7.3.ebuild [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+	KEYWORDS="amd64 ~arm64 ~riscv ~x86"
+fi
+
+LICENSE="MIT"
+SLOT="0"
+IUSE="debug +gles +system-wfconfig +system-wlroots X"
+
+DEPEND="
+	dev-libs/libevdev
+	dev-libs/libinput
```
Comment (juippis) [inline gui-wm/wayfire/wayfire-0.7.3.ebuild:25]: Why are both of these required, aren't these supposed to replace each other?

Comment (juippis) [inline gui-wm/wayfire/wayfire-0.7.3.ebuild:25]:
```suggestion
	dev-libs/libinput:=
```

File: gui-wm/wayfire/wayfire-0.7.3.ebuild [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+	dev-libs/wayland
+	gui-libs/gtk-layer-shell
+	media-libs/glm
+	media-libs/mesa:=[gles2,wayland,X?]
+	media-libs/libglvnd[X?]
+	media-libs/libjpeg-turbo
```
Comment (juippis) [inline gui-wm/wayfire/wayfire-0.7.3.ebuild:31]:
```suggestion
	media-libs/libjpeg-turbo:=
```

File: gui-wm/wayfire/wayfire-0.7.3.ebuild [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+	media-libs/libpng
```
Comment (juippis) [inline gui-wm/wayfire/wayfire-0.7.3.ebuild:32]:
```suggestion
	media-libs/libpng:=
```

File: gui-wm/wayfire/wayfire-0.7.3.ebuild [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+	media-libs/freetype:=[X?]
+	x11-libs/libdrm
+	x11-libs/gtk+:3=[wayland,X?]
+	x11-libs/cairo:=[X?,svg]
```
Comment (juippis) [inline gui-wm/wayfire/wayfire-0.7.3.ebuild:36]:
```suggestion
	x11-libs/cairo[X?,svg]
```

File: gui-wm/wayfire/wayfire-0.7.3.ebuild [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+	x11-libs/libxkbcommon:=[X?]
```
Comment (juippis) [inline gui-wm/wayfire/wayfire-0.7.3.ebuild:37]:
```suggestion
	x11-libs/libxkbcommon[X?]
```

File: gui-wm/wayfire/wayfire-0.7.3.ebuild [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+	x11-libs/pixman
```
Comment (juippis) [inline gui-wm/wayfire/wayfire-0.7.3.ebuild:38]:
```suggestion
	x11-libs/libxkbcommon[X?]
	x11-libs/pango
	x11-libs/pixman
```

File: gui-wm/wayfire/wayfire-0.7.3.ebuild (hunk 1 above)
Comment (m1027) [reply gui-wm/wayfire/wayfire-0.7.3.ebuild]: Thanks, applied, readding [X?] though.

File: gui-wm/wayfire/wayfire-0.7.3.ebuild (hunk 2 above)
Comment (m1027) [reply gui-wm/wayfire/wayfire-0.7.3.ebuild:42]: I did more USE on/off tests, after all applying suggestions and changes. +X works now.

File: gui-wm/wayfire/wayfire-0.7.3.ebuild (hunk 3 above)
Comment (m1027) [reply gui-wm/wayfire/wayfire-0.7.3.ebuild]: Applied.

File: gui-wm/wayfire/wayfire-0.7.3.ebuild (hunk 4 above)
Comment (m1027) [reply gui-wm/wayfire/wayfire-0.7.3.ebuild]: Applied.

File: gui-wm/wayfire/wayfire-0.7.3.ebuild (hunk 5 above)
Comment (m1027) [reply gui-wm/wayfire/wayfire-0.7.3.ebuild]: Applied.

Comment (m1027) [reply gui-wm/wayfire/wayfire-0.7.3.ebuild]: Applied.

File: gui-wm/wayfire/wayfire-0.7.3.ebuild (hunk 6 above)
Comment (m1027) [reply gui-wm/wayfire/wayfire-0.7.3.ebuild]: Applied

File: gui-wm/wayfire/wayfire-0.7.3.ebuild (hunk 8 above)
Comment (m1027) [reply gui-wm/wayfire/wayfire-0.7.3.ebuild]: Applied.

**Comment by m1027** [comment]
So, thanks for the help with this ebuild and the nice explanatory suggestions.

To have lssens learned I generalized your suggestens a bit and come up with a couple of questions (sightly OT). Just in case you find time to answer on that...:

(1) You suggested to add/remove ":=" for slots. Is that manual work for devs, I mean, that you check every dependency whether there are slots or not?

(2) Avoiding alternative libraries as deps (libevdev, libinput): This is probably only feasible with some experience I guess.

(3) Missing deps: You added e.g. pango, right. As this had been obviously present already on my box, how would I do a proper dependency check when working on ebuilds...

(4) USE flag on/off testing. I did that manually for now and this is a bit error prone, too, as adding +X might trigger other packages +X and one has to remove that from those deps afterwards carefully again to do proper tests... I wonder whether there is a better way to do these USE on/off build tests.

Thanks again

**Comment by thesamesam** [comment]
> So, thanks for the help with this ebuild and the nice explanatory suggestions.
> 
> To have lssens learned I generalized your suggestens a bit and come up with a couple of questions (sightly OT). Just in case you find time to answer on that...:
> 
> (1) You suggested to add/remove ":=" for slots. Is that manual work for devs, I mean, that you check every dependency whether there are slots or not?
> 

iwdevtools can help show you this!

> (2) Avoiding alternative libraries as deps (libevdev, libinput): This is probably only feasible with some experience I guess.
> 
> (3) Missing deps: You added e.g. pango, right. As this had been obviously present already on my box, how would I do a proper dependency check when working on ebuilds...

iwdevtools again :)
> 
> (4) USE flag on/off testing. I did that manually for now and this is a bit error prone, too, as adding +X might trigger other packages +X and one has to remove that from those deps afterwards carefully again to do proper tests... I wonder whether there is a better way to do these USE on/off build tests.

juippis uses https://wiki.gentoo.org/wiki/User:Juippis/The_ultimate_testing_system_with_lxd but see also https://wiki.gentoo.org/wiki/Package_testing. Lots of other tools out there: https://wiki.gen.

> 
> Thanks again

**Comment by thesamesam** [comment]
Could you as a follow up PR sync the live ebuild? Ideally use a template (if [[ ${PV} == 9999 ]] ..., like other ebuilds often do) to make it easier to keep in sync?

**Comment by thesamesam** [comment]
Merged (squashed): 043e5f8bbc1910d4188cf2d1937fcc378e39f275

**Comment by m1027** [comment]
notifications:

> Could you as a follow up PR sync the live ebuild? Ideally use a
> template (if [[ ${PV} == 9999 ]] ..., like other ebuilds often do)
> to make it easier to keep in sync?

Well, I updated wayfire-9999 according to the recent wayfire-0.7.3.
However, building wayfire-9999 fails due to some internal
meson.build issues:

> meson.build:134:2: ERROR: Dependencies must be external dependencies

While this may (or may not) be trivial to fix it is probably beyond
creating ebuilds.

So, no PR yet.

If you appreciated to see a PR for this slightly updated version of
the existing wayfire-9999.ebuild, then let me know.

Thanks

**Pull request closed without merging**